File: upl-purchase-sdk-java/src/main/java/com/unzer/paylater/model/PurchaseState.java
/**
 * Purchase API
 * @copyright Copyright (c) 2020 Unzer Pay Later
 * @license see LICENSE.TXT
 *
 * This class is based on the Unzer Pay Later OpenAPI specification, version 1.0.0.
 */
package com.unzer.paylater.model;

public enum PurchaseState {

    INITIALIZED("INITIALIZED"),
    PRECHECKED("PRECHECKED"),
    DECLINED("DECLINED"),
    AUTHORIZED("AUTHORIZED"),
    AUTHORIZATION_PENDING("AUTHORIZATION_PENDING"),
    CANCELLED("CANCELLED"),
    FULFILLMENT("FULFILLMENT"),
    BLOCKED("BLOCKED"),
    TIMED_OUT("TIMED_OUT"),
    CLOSED("CLOSED");

    private String value;

    PurchaseState(String value) {
        this.value = value;
    }

    public static PurchaseState fromValue(String value) {
        for (PurchaseState b : PurchaseState.values()) {
            if (b.value.equals(value)) {
                return b;
            }
        }
        throw new IllegalArgumentException("Unexpected value '" + value + "'");
    }

    public String getValue() {
        return value;
    }

    @Override
    public String toString() {
        return String.valueOf(value);
    }
}

File: upl-purchase-sdk-php/src/Model/PurchaseState.php
<?php

/**
 * Purchase API
 * @copyright Copyright (c) 2020 Unzer Pay Later
 * @license see LICENSE.TXT
 *
 * This class is based on the Unzer Pay Later OpenAPI specification, version 1.0.0.
 */

declare(strict_types=1);

namespace Unzer\PayLater\Model;

use InvalidArgumentException;

use function array_key_exists;
use function sprintf;

class PurchaseState implements Enum
{
    public const INITIALIZED = 'INITIALIZED';
    public const PRECHECKED = 'PRECHECKED';
    public const DECLINED = 'DECLINED';
    public const AUTHORIZED = 'AUTHORIZED';
    public const AUTHORIZATION_PENDING = 'AUTHORIZATION_PENDING';
    public const CANCELLED = 'CANCELLED';
    public const FULFILLMENT = 'FULFILLMENT';
    public const BLOCKED = 'BLOCKED';
    public const TIMED_OUT = 'TIMED_OUT';
    public const CLOSED = 'CLOSED';

    public const ALLOWED_TYPES = [
        'INITIALIZED' => self::INITIALIZED,
        'PRECHECKED' => self::PRECHECKED,
        'DECLINED' => self::DECLINED,
        'AUTHORIZED' => self::AUTHORIZED,
        'AUTHORIZATION_PENDING' => self::AUTHORIZATION_PENDING,
        'CANCELLED' => self::CANCELLED,
        'FULFILLMENT' => self::FULFILLMENT,
        'BLOCKED' => self::BLOCKED,
        'TIMED_OUT' => self::TIMED_OUT,
        'CLOSED' => self::CLOSED,
    ];

    /**
     * @var string
     */
    private $value;

    /**
      * @param string $value
      * @throws InvalidArgumentException
      */
    public function __construct(string $value)
    {
        if (!array_key_exists($value, self::ALLOWED_TYPES)) {
            throw new InvalidArgumentException(sprintf('Unexpected value "%s"', $value));
        }
        $this->value = $value;
    }

    /**
     * @return string
     */
    public function getValue(): string
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->value;
    }
}
